.github/workflows/reusable-code-quality.yml

```diff
@@ -92,4 +92,11 @@ jobs:
 
       - name: Run PHPCS
         if: steps.check_phpcs_binary_file.outputs.files_exists == 'true'
-        run: vendor/bin/phpcs -q --report=checkstyle | cs2pr
+        run: vendor/bin/phpcs -q --report=full --report-checkstyle=/tmp/phpcs-checkstyle-report.xml
+
+      - name: Convert PHPCS report to PR comments
+        if: always()
```

Comment:
Using `always()` here as the above job will fail when there are PHPCS errors and it will short-circuit further runs.

 Another thing we can try is to append `--runtime-set ignore_errors_on_exit 1 --runtime-set ignore_warnings_on_exit 1` so that the PHPCS command never fails but logs will be there. In that case `cs2pr` will fail in case of PHPCS errors/warnings.

Reply:
@thelovekesh What if we ran `vendor/bin/phpcs` twice? Once to display output in the logs, and the second time for `cs2pr`?

Reply:
@danielbachhuber In the end `cs2pr` is going to analyze the XML report so either we pipe it or pass a file, it doesn't make any difference. Adding another step seems redundant as we can achieve the same results in a single run.

Reply:
I think it's fine as it is. The `always` makes conceptual sense to me. We _always_ process the output of PHPCS regardless of whether it succeeded or failed.